### ISSUE TITLE
Using shared VPC results in incorrect modification message 

### ISSUE THREAD

**Issue by yissachar** [kubernetes/kops#476]
When creating a cluster using a shared VPC, kops thinks it will modify the names of the shared resources:

```
Will modify resources:
  VPC    vpc/k8s.somefoo.com
    Name baz -> k8s.somefoo.com

  InternetGateway    internetGateway/k8s.somefoo.com
    Name baz -> k8s.somefoo.com
```

This won't actually happen of course, so we shouldn't get this message.

**Comment by austinmoore-** [comment]
Somewhat related: when the id of an existing subnet is specified, it _does_ change the name of the subnet. I think this should be consistent with the name changing/not changing behavior of specifying an existing VPC.

**Comment by yissachar** [comment]
@austinmoore- This will be fixed once https://github.com/kubernetes/kops/pull/452 is merged (and in fact, if you test with that PR you get identical behaviour, where kops thinks the name should be changed but doesn't actually do it).

**Comment by austinmoore-** [comment]
Oh, awesome! Thanks for the heads up.

**Comment by jkemp101** [comment]
I don't believe this was fixed with the merge of #452.  I'm still getting these (alarming) messages with kops v1.4.1 using a shared VPC.  I'm not sharing subnets.

**Comment by yissachar** [comment]
@jkemp101 I wasn't saying that #452 fixes this issue, rather that it fixes the issue that kops would change the name of the subnet. Sorry for the confusion.

**Comment by itajaja** [comment]
just to be 100% sure, these messages are just false positive right? I see

![image](https://cloud.githubusercontent.com/assets/697647/19712170/980c9aa2-9b09-11e6-8ab3-2bba01102779.png)

but the first two changes will actually not be applied, correct?

**Comment by yissachar** [comment]
@justinsb Until this is fixed we should add a warning to the [existing VPC doc](https://github.com/kubernetes/kops/blob/master/docs/run_in_existing_vpc.md) that explains these changes won't actually be applied. It is a really scary message to see if you're not expecting it. (I would just submit a PR to add the warning myself but I'm still in CLA limbo...)

**Comment by chrislovecnm** [comment]
@yissachar have at the documentation update :)

**Comment by yissachar** [comment]
@chrislovecnm As mentioned, I would but it wouldn't help because my organization still hasn't signed the Linux CLA. We can wait until that happens (I don't know when it will be), or someone else can submit the relevant changes in the meantime.

**Comment by chrislovecnm** [comment]
@yissachar you cannot do work on your own? Damn dude that stinks ;(

**Comment by justinsb** [comment]
Planning to address this as part of a chunk of tagging work in 1.5.1.

**Comment by ferminromero** [comment]
Any status update on this?
Is this a particularly tough bug to fix? In other words, given a pointer to the area of code, I might be interested in taking this on.

**Comment by chrislovecnm** [comment]
I am uncertain if this is fixed, I am guessing no. Best this to do is try to recreate.

Ping us if you have questions!

**Comment by renewooller** [comment]
FYI this is not fixed on 1.7. After running 'kops update cluster  <mycluster>' and adding 'additionalPolicies' with an s3 policy, I'm getting:

```
Will create resources:
IAMRolePolicy/additional.nodes.<redacted domain>
Role                    name:nodes.<redacted domain> id:<redacted id>

Will modify resources:
DNSName/api.<redacted domain>
Lifecycle                <nil> -> Sync

DNSName/bastion.<redacted domain>
Lifecycle                <nil> -> Sync

ElasticIP/<redacted region>a.<redacted domain>
Lifecycle                <nil> -> Sync

LoadBalancer/api.<redacted domain>
Lifecycle                <nil> -> Sync

LoadBalancer/bastion.<redacted domain>
Lifecycle                <nil> -> Sync

LoadBalancerAttachment/api-master-<redacted region>a
Lifecycle                <nil> -> Sync

LoadBalancerAttachment/bastion-elb-attachment
Lifecycle                <nil> -> Sync

NatGateway/<redacted region>a.<redacted domain>
Lifecycle                <nil> -> Sync

VPC/<redacted domain>
Tags                     {KubernetesCluster: <redacted domain>, Name: <redacted domain>} -> {KubernetesCluster: <redacted domain>, Name: <redacted domain>, kubernetes.io/cluster/<redacted domain>: shared}
```

**Comment by chrislovecnm** [comment]
@renewooller yes still a work in progress

**Comment by alok87** [comment]
@chrislovecnm @justinsb anyone working on it? should i give a fix?

**Comment by KashifSaadat** [comment]
@alok87 have a peek at PR #3184 which has some progress on a fix for this issue.

**Comment by justinsb** [comment]
Apparently this may be more serious.  From slack:

> FYI, the deleting resource bug is way worse than I thought. Kops will actually delete subnets from other VPCs if the KubernetesCluster tag is present
> And ignores the `kubernetes.io/cluster/` tag completely

Putting into 1.8.0 - it got left behind in an old milestone

**Comment by mikesplain** [comment]
This is likely fixed given all the tag changes around 1.9 and soon 1.10.  Feel free to reopen if necessary!

/close
